hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -636,6 +616,34 @@ private <T> ClosableIterator<T> lookupRecords(List<String> sortedKeys,
     }
   }
 
+  private Predicate buildPrefixPredicate(String partitionName, List<String> sortedKeys) {
+    if (MetadataPartitionType.fromPartitionPath(partitionName)
+          .equals(MetadataPartitionType.SECONDARY_INDEX)) {
+      return Predicates.startsWithAny(null,
+          sortedKeys.stream()
+          .map(SecondaryIndexKeyUtils::constructSecondaryIndexKeyPrefix)
+          .map(Literal::from)
+          .collect(Collectors.toList()));
+    }
+    return Predicates.startsWithAny(null, sortedKeys.stream()
+        .map(Literal::from)
+        .collect(Collectors.toList()));
+  }
```

Comment:
Could we have one `buildPredicate` method and consider both scenarios of full keys and prefix keys instead of separate methods?